mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+
+    return data
+
+
+def _interpolate_data(stc, morph_data, mri_resolution=True, mri_space=True,
+                      format='nifti1'):
+    """Interpolate source estimate data to MRI."""
+    if format != 'nifti1' and format != 'nifti2':
+        raise ValueError("invalid format specifier %s. Must be 'nifti1' or"
+                         " 'nifti2'" % format)
+    if format == 'nifti1':
+        from nibabel import (Nifti1Image as NiftiImage,
+                             Nifti1Header as NiftiHeader)
+    elif format == 'nifti2':
+        from nibabel import (Nifti2Image as NiftiImage,
+                             Nifti2Header as NiftiHeader)
+    from dipy.align.reslice import reslice
+
+    # setup volume parameters
+    n_times = stc.data.shape[1]
+    shape3d = morph_data['src_shape']
+    shape = (n_times,) + shape3d
+    vol = np.zeros(shape)
+
+    voxel_size_defined = False
+
+    if isinstance(mri_resolution, (int, float)) and not isinstance(
+            mri_resolution, bool):
+        # use iso voxel size
+        mri_resolution = (float(mri_resolution),) * 3
+
+    if isinstance(mri_resolution, tuple):
+        voxel_size = mri_resolution
+        voxel_size_defined = True
+        mri_resolution = True
+
+    # if data wasn't morphed yet - necessary for call of
+    # stc_unmorphed.as_volume. Since only the shape of src is known, it cannot
+    # be resliced to a given voxel size without knowing the original.
+    if 'morph_shape' not in morph_data and voxel_size_defined:
+        raise ValueError(
+            "Cannot infer original voxel size for reslicing... "
+            "set mri_resolution to boolean value or apply morph first.")
+
+    # use mri resolution as represented in src
+    if mri_resolution:
+        mri_shape3d = morph_data['src_shape_full']
+        mri_shape = (n_times,) + mri_shape3d
+        mri_vol = np.zeros(mri_shape)
+        interpolator = morph_data['interpolator']
+
+    mask3d = morph_data['inuse'].reshape(shape3d).astype(np.bool)
+    n_vertices = np.sum(mask3d)
+
+    n_vertices_seen = 0
+    for k, v in enumerate(vol):  # loop over time instants
+        stc_slice = slice(n_vertices_seen, n_vertices_seen + n_vertices)
+        v[mask3d] = stc.data[stc_slice, k]
+
+    n_vertices_seen += n_vertices
+
+    if mri_resolution:
+        for k, v in enumerate(vol):
+            mri_vol[k] = (interpolator * v.ravel()).reshape(mri_shape3d)
+        vol = mri_vol
+
+    vol = vol.T
+
+    # set correct space
+    affine = morph_data['src_affine_vox']
+
+    if not mri_resolution:
+        affine = morph_data['src_affine_src']
+
+    if mri_space:
+        affine = np.dot(morph_data['src_affine_ras'], affine)
+
+    affine[:3] *= 1e3
+
+    # pre-define header
+    header = NiftiHeader()
+    header.set_xyzt_units('mm', 'msec')
+    header['pixdim'][4] = 1e3 * stc.tstep
+
+    with warnings.catch_warnings(record=True):  # nibabel<->numpy warning
```

Comment:
this catches all warnings, maybe you want to check for this `nibabel<->numpy warning` and let the rest pass.

I'm not sure if there's a way to do something like this:
```py
with warnings.catch_warnings():
     warnings.simplefilter('ignore', XXXWarning)
     img = NiftiImage(vol, affine, header=header)
```
but that it also allows for some regex, to make sure that you catch what you want but not somethings else.